### DISCUSSION TITLE
1.35.1

### DISCUSSION THREAD

**Discussion by Koenkk** [Koenkk/zigbee2mqtt#20637] [Releases]
Feel free to discuss the [v1.35.1](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.35.1) release here!

**Comment by pyrodex** [comment]
Seeing the following error in HA/Z2M now after this update with a Xiaomi WS-USC04 (https://www.zigbee2mqtt.io/devices/WS-USC04.html#xiaomi-ws-usc04).

![image](https://github.com/Koenkk/zigbee2mqtt/assets/108703/082584f8-f82a-44c9-9bd9-e9e68f5210bc)

Here are the debug logs from Z2M for reference. I did not see this with Z2M 1.35.0.

EDIT: Guessing its tied to this? https://github.com/Koenkk/zigbee-OTA/commit/1122f56eefc81dd943804d0e8a8c5585b48285cb

```
debug 2024-01-07 10:30:57: Received MQTT message on 'zigbee2mqtt/bridge/request/device/ota_update/check' with data '{"id":"Deck dual light switch","transaction":"qyq1l-2"}'
info  2024-01-07 10:30:57: Checking if update available for 'Deck dual light switch'
debug 2024-01-07 10:30:57: OTA: Checking if update available for '0x54ef441000990210' (lumi.switch.b2naus01)
debug 2024-01-07 10:30:57: OTA: Using endpoint '1'
debug 2024-01-07 10:30:58: Received Zigbee message from 'Deck dual light switch', type 'commandQueryNextImageRequest', cluster 'genOta', data '{"fieldControl":0,"fileVersion":31,"imageType":404,"manufacturerCode":4447}' from endpoint 1 with groupID 0
info  2024-01-07 10:30:58: MQTT publish: topic 'zigbee2mqtt/Deck dual light switch', payload '{"consumption":0.1621781885623932,"current":0,"device":{"applicationVersion":31,"dateCode":"12-28-2021","friendlyName":"Deck dual light switch","hardwareVersion":1,"ieeeAddr":"0x54ef441000990210","manufacturerID":4447,"manufacturerName":"LUMI","model":"WS-USC04","networkAddress":25112,"powerSource":"Mains (single phase)","stackVersion":2,"type":"Router","zclVersion":3},"device_temperature":24,"energy":0.16,"flip_indicator_light":"ON","last_seen":"2024-01-07T10:30:58-05:00","linkquality":142,"operation_mode_bottom":"control_relay","operation_mode_top":"control_relay","power":0,"power_outage_count":0,"power_outage_memory":true,"state":"OFF","state_bottom":"OFF","state_top":"OFF","update":{"installed_version":31,"latest_version":-1,"state":"idle"},"update_available":null,"voltage":122}'
debug 2024-01-07 10:30:58: OTA: Got request '{"fieldControl":0,"manufacturerCode":4447,"imageType":404,"fileVersion":31}'
debug 2024-01-07 10:30:58: OTA: Is new image available for '0x54ef441000990210', current '{"fieldControl":0,"manufacturerCode":4447,"imageType":404,"fileVersion":31}'
debug 2024-01-07 10:30:58: ZigbeeOTA: call getImageMeta for lumi.switch.b2naus01
debug 2024-01-07 10:30:58: ZigbeeOTA: downloaded main index
warn  2024-01-07 10:30:58: OTA: Images currently unavailable for device 'lumi.switch.b2naus01', hardwareVersion '1', manufacturerName LUMI, {"fieldControl":0,"manufacturerCode":4447,"imageType":404,"fileVersion":31}'
debug 2024-01-07 10:30:58: OTA: Update available for '0x54ef441000990210': NO
info  2024-01-07 10:30:58: No update available for 'Deck dual light switch'
info  2024-01-07 10:30:58: MQTT publish: topic 'zigbee2mqtt/Deck dual light switch', payload '{"consumption":0.1621781885623932,"current":0,"device":{"applicationVersion":31,"dateCode":"12-28-2021","friendlyName":"Deck dual light switch","hardwareVersion":1,"ieeeAddr":"0x54ef441000990210","manufacturerID":4447,"manufacturerName":"LUMI","model":"WS-USC04","networkAddress":25112,"powerSource":"Mains (single phase)","stackVersion":2,"type":"Router","zclVersion":3},"device_temperature":24,"energy":0.16,"flip_indicator_light":"ON","last_seen":"2024-01-07T10:30:58-05:00","linkquality":142,"operation_mode_bottom":"control_relay","operation_mode_top":"control_relay","power":0,"power_outage_count":0,"power_outage_memory":true,"state":"OFF","state_bottom":"OFF","state_top":"OFF","update":{"installed_version":31,"latest_version":-1,"state":"idle"},"update_available":null,"voltage":122}'
info  2024-01-07 10:30:58: MQTT publish: topic 'zigbee2mqtt/bridge/response/device/ota_update/check', payload '{"data":{"id":"Deck dual light switch","updateAvailable":false},"status":"ok","transaction":"qyq1l-2"}'
```

**Reply by Nerivec** [reply]
That's with the new checks on OTA images (combined with the commit you mentioned). It now makes the distinction between "no update available", and "no image available". 
I believe that issue is happening on the HA side. AwesomeVersion doesn't seem to support [negative versions](https://github.com/home-assistant/core/blob/3139e926965b464c1172770830a4ab3ae6283959/homeassistant/components/update/__init__.py#L190)... and HA [defaults to "ON"](https://github.com/home-assistant/core/blob/3139e926965b464c1172770830a4ab3ae6283959/homeassistant/components/update/__init__.py#L411) on failed compare... I guess we could return `installed_version` in `latest_version` if `no image available`, but I'll let @Koenkk decide (there are other [instances](https://github.com/Koenkk/zigbee2mqtt/blob/9eaaa0f9d1725416639ee894fc45542be3a4753f/lib/extension/homeassistant.ts#L1736) where Z2M returns -1 that would also fail it seems). I think it definitely would be better if HA handled that better though.

**Reply by Nerivec** [reply]
In any case, I PRed HA core https://github.com/home-assistant/core/pull/107483 ...

**Reply by Koenkk** [reply]
>  I guess we could return installed_version in latest_version if no image available

In addition to the HA pr this sounds like better behaviour to me

**Comment by ralfw999** [comment]
Seems after the update from 1.35.0 which worked fine to 1.35.1 a different Version of index.js required.
zigbe2mqtt is restarting every 3 seconds now
Installed in OpenHAB 3.1.0

Jan  7 20:08:45 openhabian npm[21842]: > zigbee2mqtt@1.35.1 start
Jan  7 20:08:45 openhabian npm[21842]: > node index.js
Jan  7 20:08:46 openhabian npm[21842]: #011#011Zigbee2MQTT requires node version ^18 || ^20 || ^21, you are running v16.19.1!
Jan  7 20:08:46 openhabian npm[21842]: Error: Cannot find module './dist/util/settings'
Jan  7 20:08:46 openhabian npm[21842]: Require stack:
Jan  7 20:08:46 openhabian npm[21842]: - /opt/zigbee2mqtt/index.js
Jan  7 20:08:46 openhabian npm[21842]:     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1021:15)
Jan  7 20:08:46 openhabian npm[21842]:     at Function.Module._load (node:internal/modules/cjs/loader:866:27)
Jan  7 20:08:46 openhabian npm[21842]:     at Module.require (node:internal/modules/cjs/loader:1093:19)
Jan  7 20:08:46 openhabian npm[21842]:     at require (node:internal/modules/cjs/helpers:108:18)
Jan  7 20:08:46 openhabian npm[21842]:     at start (/opt/zigbee2mqtt/index.js:90:22)
Jan  7 20:08:46 openhabian npm[21842]:     at processTicksAndRejections (node:internal/process/task_queues:96:5)
Jan  7 20:08:46 openhabian systemd[1]: zigbee2mqtt.service: Main process exited, code=exited, status=1/FAILURE
Jan  7 20:08:46 openhabian systemd[1]: zigbee2mqtt.service: Failed with result 'exit-code'.
Jan  7 20:08:46 openhabian systemd[1]: zigbee2mqtt.service: Service RestartSec=100ms expired, scheduling restart.
Jan  7 20:08:46 openhabian systemd[1]: zigbee2mqtt.service: Scheduled restart job, restart counter is at 2989.
Jan  7 20:08:46 openhabian systemd[1]: Stopped Zigbee2MQTT.
Jan  7 20:08:46 openhabian systemd[1]: Started Zigbee2MQTT.
```

**Reply by Nerivec** [reply]
I don't know OpenHAB (or how Z2M is installed there), but following the logs:

>Zigbee2MQTT requires node version ^18 || ^20 || ^21, you are running v16.19.1!

Seems your machine is using an old version of node (I see in OpenHAB repo they seem to be using [v18](https://github.com/openhab/openhabian/blob/9ac4ddd2ed5b747f8af0994c2aa79f107323eb90/functions/nodejs-apps.bash#L12)). I'd start by updating that and then trying Z2M update again. Since it's failing on a missing file, I'd guess it's not compiling typescript properly.

**Reply by ralfw999** [reply]
This is correct, but I failed to update the node version. What helped in the end was deinstalling zigbee2mqtt and reinstalling it.
Do not forget to make a BACKUP of the data folder in /opt/zigbee2mqtt upfront!!!!
Surprisingly I still have an old Version of node.js
openhabian@openhabian:~ $ node -v
v16.20.2

**Comment by michalw2-alt** [comment]
I'm still getting following errors in the log (HomeBridge keeps rebooting in the loop):

```
[07/01/2024, 22:39:38] [zigbee2mqtt] Zigbee2MQTT is ONLINE
[07/01/2024, 22:39:39] [zigbee2mqtt] New accessory: Coordinator
[07/01/2024, 22:39:39] [zigbee2mqtt] Failed to process MQTT message on 'zigbee2mqtt/bridge/devices'. (Maybe check the MQTT version?)
[07/01/2024, 22:39:39] [zigbee2mqtt] Cannot read properties of null (reading 'vendor')

[07/01/2024, 22:41:45] TypeError: Cannot read properties of null (reading 'vendor')
    at isDeviceDefinition (/var/lib/homebridge/node_modules/homebridge-z2m/src/z2mModels.ts:302:72)
    at Zigbee2mqttAccessory.updateDeviceInformation (/var/lib/homebridge/node_modules/homebridge-z2m/src/platformAccessory.ts:404:30)
    at new Zigbee2mqttAccessory (/var/lib/homebridge/node_modules/homebridge-z2m/src/platformAccessory.ts:97:10)
    at Zigbee2mqttPlatform.addAccessory (/var/lib/homebridge/node_modules/homebridge-z2m/src/platform.ts:562:19)
    at Zigbee2mqttPlatform.configureAccessory (/var/lib/homebridge/node_modules/homebridge-z2m/src/platform.ts:461:10)
    at /var/lib/homebridge/node_modules/homebridge/src/bridgeService.ts:328:25
    at Array.filter (<anonymous>)
    at BridgeService.restoreCachedPlatformAccessories (/var/lib/homebridge/node_modules/homebridge/src/bridgeService.ts:288:69)
    at Server.start (/var/lib/homebridge/node_modules/homebridge/src/server.ts:188:24)
[07/01/2024, 22:41:45] Got SIGTERM, shutting down Homebridge...
```
```
[07/01/2024, 22:41:45] AssertionError [ERR_ASSERTION]: Cannot generate setupURI on an accessory that isn't published yet!
    at Bridge.Accessory.setupURI (/var/lib/homebridge/node_modules/homebridge/node_modules/hap-nodejs/src/lib/Accessory.ts:994:11)
    at Server.setServerStatus (/var/lib/homebridge/node_modules/homebridge/src/server.ts:155:45)
    at Server.teardown (/var/lib/homebridge/node_modules/homebridge/src/server.ts:199:10)
    at signalHandler (/var/lib/homebridge/node_modules/homebridge/src/cli.ts:93:12)
    at process.emit (node:events:514:28)
    at process.emit (/var/lib/homebridge/node_modules/homebridge/node_modules/source-map-support/source-map-support.js:516:21)
```
Last working version for me is 1.34.0 (no "New accessory: Coordinator" log entry is present in 1.34.0).

**Reply by Nerivec** [reply]
I think your homebridge-z2m package needs an update (you log line numbers don't match with current code).
And also: https://github.com/itavero/homebridge-z2m/releases/tag/v1.9.3

**Reply by michalw2-alt** [reply]
Thank you. That solved the problem. Much appreciated.

**Reply by mistressxalexis** [reply]
♦️

**Comment by merlinpimpim** [comment]
Well done!
The most reliable version for weeks.
I haven't had to restart Z2M for about 8 consecutive hours - that hadn't happened for weeks!

**Reply by lux73** [reply]
never had any Problems that would  require frequently z2m restarts

**Reply by merlinpimpim** [reply]
Lucky man!
https://github.com/Koenkk/zigbee2mqtt/issues/19168
https://github.com/Koenkk/zigbee2mqtt/issues/20519
https://github.com/Koenkk/zigbee2mqtt/issues/20253

https://github.com/Koenkk/zigbee2mqtt/discussions/20637#discussioncomment-8039906
https://github.com/Koenkk/zigbee2mqtt/discussions/20637#discussioncomment-8041113

**Reply by lux73** [reply]
sometimes People are unable to read properly what's to do ->

`Jan 7 20:08:46 openhabian npm[21842]: https://github.com/Koenkk/zigbee2mqtt/issues/11#011Zigbee2MQTT requires node version ^18 || ^20 || ^21, you are running v16.19.1!`

there is exactly described what to to: NodeJS v16 is outdatet -> you have to update to your NodeJS to v18 at least ^^

**Reply by merlinpimpim** [reply]
Which of my comments does this relate to?

**Reply by lux73** [reply]
your second last Link ^^

https://github.com/Koenkk/zigbee2mqtt/discussions/20637#discussioncomment-8039906

**Reply by merlinpimpim** [reply]
ok, thank.
But I'm not affected by this particular problem (20637).
I just gave you examples of people who have to restart z2m regularly (because you wrote that you didn't have this problem in response to my initial comment).

**Reply by ralfw999** [reply]
The post is from me, and believe it or not I am able to read what to do. 
However:
1. I did not expect significant Version conflicts by updating from 1.35.0 to 1.35.1
2. Such hints and solutions from other users helped me for others issues a lot in the past. That's also why I posted my problem. 
3. Updating NodeJS to a higher Version was (at least for me) not trivial / did not work in my openHAB environment
4. After de- and reinstalling z2m 1.35.1 is working fine with NodeJS v16.20.2!!!
5. I am very happy with z2m and like it a lot. So I wanted to contribute.

**Comment by lux73** [comment]
i am using plain z2m @ Debian Bare Metal with Slea.sh CC2652RB Stick (FW 20230922) - not ZHA or any Docker implementation...  don't know but my z2m is really really stable and snappy

restart only for regulary Updates

**Reply by merlinpimpim** [reply]
Congratulations

**Reply by lux73** [reply]
no need to be sarcastic

**Comment by Burki24** [comment]
@Koenkk 

After update from 1.34.xx to 1.35.1 there are some problems to get the exposes of a device. Normaly, if I send a request to bridge (like described here: https://www.zigbee2mqtt.io/guide/usage/mqtt_topics_and_messages.html#zigbee2mqtt-bridge-devices) there are devices showing their exposes. But other devices do not show their exposes. 

![image](https://github.com/Koenkk/zigbee2mqtt/assets/20765982/26867320-20c3-4c3c-b382-35386d7e2f36)
Old style

![image](https://github.com/Koenkk/zigbee2mqtt/assets/20765982/db22cd79-0cb4-4174-b9b4-244c985fbaab)
New style


I tried to find out the difference and it looks like the classical converter-form 'exposes:[];' will run and bring up all exposes from a device. If the converter uses 'exposes: (device, options) => {const exps: Exposes [] ....' they don't send their exposes within 'zigbee2mqtt/bridge/devices'. So how will it be possible to get all exposes from a device in the future?

Many Thanks for your suport.

**Reply by Koenkk** [reply]
Both should work, can you create a new issue in the z2m issue tracker with the wrong `bridge/devices` message?

**Reply by Burki24** [reply]
@Koenkk you can find it here: https://github.com/Koenkk/zigbee2mqtt/issues/21083

Thanks a lot for your help.

**Comment by Jk607617627** [comment]
Hi everyone, I am using zigbee2mqtt on my QNAP NAS. Why I am not able update it from version 
![Snímek obrazovky 2024-01-31 194945](https://github.com/Koenkk/zigbee2mqtt/assets/106123485/2a8e36f9-43fb-4268-b7e5-83ec12b1b498)
to the newest? When I am trying to pull the newer version, it doesnt offer me any new
![Snímek obrazovky 2024-01-31 195217](https://github.com/Koenkk/zigbee2mqtt/assets/106123485/9850ad35-2a98-430f-9394-1d82100380bc)
